### ISSUE TITLE
Cosine Penalty for first line never used?

### ISSUE THREAD

**Issue by gemsanyu** [longhoangphi225/MultiSample-Hypernetworks#1]
https://github.com/longhoangphi225/MultiSample-Hypernetworks/blob/17aca2ccab6e905d02b9b2bd1e85395007da792b/Jura/trainer.py#L158-L160

We see in this block that penalty[0] term in `direction_loss` will always be removed?
is there a reason? or is it an error?

**Comment by longhoangphi225** [comment]
Thank you so much. This is my coding mistake. It should be direction_loss = 0, you can fix it. However, this mistake is only in the warm-up step of our framework, so it does not have a serious impact to final results.
Cheers!
